server/server.js

```diff
@@ -54,8 +55,24 @@ app.use(methodOverride());
 // LoopBack REST interface
 app.use(app.get('restApiRoot'), loopback.rest());
 
-// API explorer
-require('loopback-component-explorer')(app);
+// To prevent the snyk warning about swagger-ui@2.x, the explorer component
+// is moved from dependencies to devDependencies.
+// Here we try to mount the explorer component in case it's not installed.
+
+try {
+  require('loopback-component-explorer')(app);
+} catch (e) {
+  const notFound = e.code === 'MODULE_NOT_FOUND' && e.message;
```

Comment:
I think you forgot to check that `e.message` is referring to explorer? I think the expression `e.message` is always truthy.

On the second thought, using `require.resolve` may be a simpler option. Quoting from https://nodejs.org/api/modules.html#modules_require_resolve_request_options

> Use the internal require() machinery to look up the location of a module, but rather than loading the module, just return the resolved filename.
> 
> If the module can not be found, a `MODULE_NOT_FOUND` error is thrown.

```js
let hasExplorer;
try {
  require.resolve('loopback-component-explorer');
  hasExplorer = true;
} catch(err) {
  hasExplorer = false;
}

if (hasExplorer) {
  require('loopback-component-explorer')(app);
}
```

Reply:
@bajtos I think, if I require a certain module and get error "MODULE_NOT_FOUND", then the only possibility is that particular module not found, isn't it?

Reply:
hmm, true `require.resolve('loopback-component-explorer');` sounds better